libraries/type-length-value/src/discriminator.rs

```diff
@@ -0,0 +1,54 @@
+//! Discriminator for differentiating account types, the "Type" in the
+//! Type-Length-Value structure. Since the word "type" is reserved in Rust,
+//! we use the term "Discriminator" and "Type" interchangeably.
```

Comment:
Yeah sure, where do you want that documentation exactly? I'm not sure it belongs in discriminator.rs since that's just the discriminator. Maybe in the `TlvState` trait?

Reply:
Added to `TlvState` trait, let me know what you think!